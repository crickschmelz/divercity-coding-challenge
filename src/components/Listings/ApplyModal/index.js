import React, { useState } from "react";
import {
  Modal,
  Button,
  Form,
  Input,
  ExitButton,
  SuccessModal,
  RegisterHeader
} from "./style";
import { applyToJob } from "../../utils";

const ApplyModal = (props) => {
  const { closeApplication } = props;
  const [motivation, setMotivation] = useState("");
  const [cover_letter, setCoverLetter] = useState("");
  const [showMessage, setShowMessage] = useState(false);

  const openMessage = (e) => {
    setShowMessage(true);
    setTimeout(() => {
      setShowMessage(false);
      closeApplication();
    }, 2000);
    e.currentTarget.reset();
  };

  const apply = (e) => {
    e.preventDefault();
    const token = localStorage.getItem("token");
    applyToJob("https://divercity-test.herokuapp.com/jobs/2/apply", token, {
      motivation,
      cover_letter
    }).then((data) => {
      return data;
    });
    openMessage(e);
  };

  return (
    <Modal>
      <ExitButton onClick={closeApplication}>x</ExitButton>
      <RegisterHeader>Apply</RegisterHeader>
      {showMessage && (
        <SuccessModal>
          <p>Success! You've applied to this job.</p>
        </SuccessModal>
      )}
      <Form onSubmit={apply} enctype="application/x-www-form-urlencoded">
        <Input
          type="textarea"
          required
          placeholder="Why do you want the job?"
          onChange={(event) => setMotivation(event.target.value)}
        ></Input>
        <Input
          type="textarea"
          required
          placeholder="Anything else to add? (i.e. cover letter)"
          onChange={(event) => setCoverLetter(event.target.value)}
        ></Input>
        <Button type="submit">Submit</Button>
      </Form>
    </Modal>
  );
};

export default ApplyModal;
