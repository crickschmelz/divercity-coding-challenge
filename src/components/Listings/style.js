import styled from "styled-components";

export const Form = styled.form`
  display: grid;
  justify-content: center;
  align-items: center;
`;

export const Input = styled.input`
  width: 300px;
  padding: 5px;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: 1px solid #bbb;
`;

export const Nav = styled.nav`
  background-color: white;
  height: auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
  width: 100%;
`;
export const NavLinkContainer = styled.ul`
  display: flex;
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
  justify-content: flex-end;
  align-items: center;
`;

export const NavLink = styled.button`
  font-size: 16px;
  color: rgb(37, 55, 85);
  margin: 10px 20px;
  text-decoration: none;
  cursor: pointer;
  border: none;
  font-weight: bold;
  background-color: transparent;

  &:after {
    content: "";
    width: 0px;
    height: 1.5px;
    display: block;
    background: hsl(346, 85%, 0%);
    transition: 300ms;
  }

  &:hover:after {
    width: 100%;
  }
`;

export const PageTitle = styled.h2`
  text-align: center;
`;
