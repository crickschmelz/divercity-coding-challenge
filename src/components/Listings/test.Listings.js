import React from "react";
import Listings from "./Listings";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Listings />).toJSON();
  expect(tree).toMatchSnapshot();
});
