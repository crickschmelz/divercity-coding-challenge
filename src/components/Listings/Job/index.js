import React from "react";
import PropTypes from "prop-types";
import { Header, ApplyButton, Div } from "./style";

const Job = (props) => {
  Job.propTypes = {
    openApplication: PropTypes.func,
    title: PropTypes.string,
    description: PropTypes.string,
    location: PropTypes.string,
    company: PropTypes.string
  };

  const { openApplication, title, description, location, company } = props;

  function createMarkup() {
    return { __html: description };
  }

  return (
    <Div>
      <Header>
        {company} - {title}
      </Header>
      <h4>{location}</h4>
      <div dangerouslySetInnerHTML={createMarkup()} />
      <ApplyButton onClick={openApplication}>Apply</ApplyButton>
    </Div>
  );
};

export default Job;
