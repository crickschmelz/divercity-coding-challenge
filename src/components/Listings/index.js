import React, { useState, useEffect } from "react";
import Job from "./Job";
import ApplyModal from "./ApplyModal/";
import {
  PageTitle,
  Nav,
  NavLinkContainer,
  NavLink,
  Form,
  Input
} from "./style";
import { getJobs } from "../utils";

const Listings = (props) => {
  const [jobList, setJobList] = useState([]);
  const [showModal, setShowModal] = useState(false);
  const [filter, setFilter] = useState("");

  useEffect(() => {
    const loadJobs = async () => {
      const { jobs } = await getJobs();
      setJobList(jobs);
    };
    loadJobs();
  }, []);

  const filterJobs = (list) => {
    if (filter === "") {
      return list;
    }
    const keys = ["location", "skills_tag", "job_type"];
    return list.filter((job) =>
      keys.some((key) => job[key] && job[key].includes(filter))
    );
  };

  const filteredList = filterJobs(jobList);

  const searchJobs = (e) => {
    e.preventDefault();
    const param = e.target.value;
    setFilter(param);
  };

  const logOut = () => {
    props.history.push("/");
    localStorage.clear();
  };

  const openApplication = () => {
    const token = localStorage.getItem("token");
    !token || token === null ? setShowModal(false) : setShowModal(true);
  };

  const closeApplication = () => {
    setShowModal(!showModal);
  };

  return (
    <main>
      {showModal && <ApplyModal closeApplication={closeApplication} />}
      <Nav>
        <a href="https://www.divercity.io/">
          <img
            src="https://www.divercity.io/static/media/complete-logo.f047ae08.svg"
            alt="divercity logo"
          ></img>
        </a>
        <NavLinkContainer>
          <li>
            <NavLink onClick={logOut}>Log Out</NavLink>
          </li>
        </NavLinkContainer>
      </Nav>
      <PageTitle>Job Listings</PageTitle>
      <Form>
        <Input
          type="text"
          required
          placeholder="Filter by location, job type, or skill..."
          onChange={searchJobs}
        ></Input>
      </Form>
      {filteredList.length ? (
        filteredList.map((job) => (
          <Job
            openApplication={openApplication}
            key={job.id}
            title={job.title}
            company={job.company}
            description={job.description}
            location={job.location}
          />
        ))
      ) : (
        <p>Loading jobs...</p>
      )}
    </main>
  );
};

export default Listings;
