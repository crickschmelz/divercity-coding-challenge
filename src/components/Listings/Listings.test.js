import React from "react";
import Listings from "./";
import { renderer } from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Listings />).toJSON();
  expect(tree).toMatchSnapshot();
});
