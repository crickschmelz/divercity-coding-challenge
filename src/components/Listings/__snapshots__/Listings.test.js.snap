// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  <nav
    className="sc-fznZeY kyWsdB"
  >
    <a
      href="https://www.divercity.io/"
    >
      <img
        alt="divercity logo"
        src="https://www.divercity.io/static/media/complete-logo.f047ae08.svg"
      />
    </a>
    <ul
      className="sc-fzokOt LzftF"
    >
      <li>
        <button
          className="sc-fzqBZW dEtAJr"
          onClick={[Function]}
        >
          Log Out
        </button>
      </li>
    </ul>
  </nav>
  <h2
    className="sc-fzqNJr bYnXfq"
  >
    Job Listings
  </h2>
  <form
    className="sc-fznyAO KcYRM"
  >
    <input
      className="sc-fznKkj kxCHgt"
      onChange={[Function]}
      placeholder="Filter by location, job type, or skill..."
      required={true}
      type="text"
    />
  </form>
  <p>
    Loading jobs...
  </p>
</main>
`;
